substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/logging/JfrLogConfiguration.java

```diff
@@ -0,0 +1,161 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.jfr.logging;
+
+import com.oracle.svm.core.log.Log;
+
+import jdk.jfr.internal.LogLevel;
+
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Handles configuration for flight recorder logging.
+ */
+public enum JfrLogConfiguration {
+    INSTANCE;
+
+    private boolean loggingEnabled = false;
+    private JfrLogSelection[] selections;
+
+    public boolean shouldLog(int tagSetId, int level) {
+        if (!loggingEnabled) {
+            return false;
+        }
+        Optional<LogLevel> tagSetLogLevel = JfrLogTagSet.fromTagSetId(tagSetId).getLevel();
+        // LogLevel#level is not accessible, so we have to use ordinal + 1
+        return tagSetLogLevel.isEmpty() ? false : tagSetLogLevel.get().ordinal() + 1 <= level;
+    }
+
+    public void parse(String config) {
+        if (config.isBlank()) {
+            return;
+        } else if (config.toUpperCase().equals("HELP")) {
+            printHelp();
+            System.exit(0);
+        }
+
+        loggingEnabled = true;
+
+        String[] splitConfig = config.split(",");
+        selections = new JfrLogSelection[splitConfig.length];
+
+        int index = 0;
+        for (String s : splitConfig) {
+            JfrLogSelection selection = new JfrLogSelection();
+            selection.parse(s);
+            selections[index++] = selection;
+        }
+        setLogTagSetLevels();
+        verifySelections();
+    }
+
+    private void setLogTagSetLevels() {
+        for (JfrLogTagSet tagSet : JfrLogTagSet.values()) {
+            Optional<LogLevel> level = Optional.empty();
+            for (JfrLogSelection selection : selections) {
+                if ((selection.wildcard && tagSet.getTags().containsAll(selection.tags))
+                        || (selection.tags.equals(tagSet.getTags()))) {
+                    level = Optional.of(selection.level);
+                    selection.matchesATagSet = true;
+                }
+            }
+            tagSet.setLevel(level);
+        }
+    }
+
+    private void verifySelections() {
+        for (JfrLogSelection selection : selections) {
+            if (!selection.matchesATagSet) {
+                Log.log().string("error: No tag set matches tag combination ")
+                    .string(selection.tags.toString().toLowerCase()).string(selection.wildcard ? "*" : "")
+                    .string(" for FlightRecorderLogging").newline();
+                System.exit(1);
+            }
+        }
+    }
+
+    private class JfrLogSelection {
+        private Set<JfrLogTag> tags = EnumSet.noneOf(JfrLogTag.class);
+        private LogLevel level = LogLevel.INFO;
+        private boolean wildcard = false;
+        private boolean matchesATagSet = false;
+
+        private void parse(String str) {
```

Comment:
It might be nicer to make this method static and to return a new object as the parsing result. Then, most of the fields could be final.